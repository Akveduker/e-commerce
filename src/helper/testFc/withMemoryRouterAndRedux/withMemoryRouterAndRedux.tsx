import { configureStore } from "@reduxjs/toolkit"
import { render } from "@testing-library/react"
import { InitialEntry } from "history"
import { Provider } from "react-redux"
import { MemoryRouter } from "react-router-dom"
import { mockReducersForStore } from "../../mock/mockStore/mockStore"
export const withMemoryRouterAndRedux = (children: React.ReactNode, reducers?: Partial<typeof mockReducersForStore>, initialEntries?: InitialEntry[]) => {
    const mockStore = configureStore({
        reducer: { ...mockReducersForStore, ...reducers }
    });
    return render(
        <MemoryRouter initialEntries={initialEntries}>
            <Provider store={mockStore}>

                {children}
            </Provider>
        </MemoryRouter>
    )
}