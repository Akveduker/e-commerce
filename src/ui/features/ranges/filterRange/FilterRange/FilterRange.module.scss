@import "../../../../../styles/_vars.scss";
@import "../../../../../styles/_mixins.scss";
.title {
    @include fontSemibold;
    margin-bottom: 16px;
}

.label {
    max-width: 109px;
}
.inputs {
    display: flex;
    align-items: flex-end;
    justify-content: space-between;
    margin-top: 15px;
    position: relative;
}
.line {
    font-size: 14px;
    color: $C1C;
    margin-bottom: 12px;
}
.label {
    &__title {
        @include fontSemibold;
    }
}
