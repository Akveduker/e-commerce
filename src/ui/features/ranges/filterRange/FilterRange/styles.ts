const dotParameters = 20
const containerMaxWidth = 266
const dotMargin = dotParameters / 2
const C2A = '#6a983c'
const C1E = '#EBEBEB'
const C1D = '#D1D1D1'
export const containerStyle: React.CSSProperties = {
    display: 'flex',
    alignItems: 'center',
    maxWidth: `${containerMaxWidth - dotParameters}px`,
    margin: `0 ${dotMargin}px`,
}
export const railStyle: React.CSSProperties = {
    backgroundColor: C1E,
    borderRadius: '12px',
    height: '6px',
    width: `${containerMaxWidth}px`,
    left: `-${dotMargin}px`
}
export const activeRailStyle: React.CSSProperties = {
    backgroundColor: C2A,
    height: '6px',
}
export const activeDotStyle: React.CSSProperties = {
    boxShadow: 'none',
}
export const dotStyle: React.CSSProperties = {
    backgroundColor: '#fff',
    height: `${dotParameters}px`,
    width: `${dotParameters}px`,
    opacity: '1',
    border: `1px solid ${C1D}`,
    marginTop: 0,
    boxShadow: '0px 2px 4px rgba(0, 0, 0, 0.15)',
}
export const styles = {
    style: containerStyle,
    railStyle: railStyle,
    trackStyle: activeRailStyle,
    handleStyle: dotStyle,
    activeDotStyle: activeDotStyle,
}