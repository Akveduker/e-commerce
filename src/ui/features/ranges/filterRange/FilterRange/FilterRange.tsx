import { FC, memo, useEffect } from 'react';
import Range, { SliderProps } from 'rc-slider';
import s from './FilterRange.module.scss'
import 'rc-slider/assets/index.css';
import { styles } from './styles';
import { RangeContext } from '../../../../../context/categories/range/RangeContext';
import { useChangeSliderValue } from '../../../../../hooks/categories/filters/ranges/useChangeSliderValue';
import { useCreateRangeContext } from '../../../../../hooks/categories/filters/ranges/useCreateRangeContext';
import { useAppSelector } from '../../../../../store/store';
import { stringFirstLetterUppercase } from '../../../../../utils/general/stringFirstLetterUppercase';
import FilterRangeInput from '../FilterRangeInput/FilterRangeInput';
import { IFilterRange, RangeNames } from '../../../../../types/filter/range/range';
import { useParams } from 'react-router-dom';
interface FilterSliderPropst extends SliderProps {
    name: RangeNames,
}
const FilterRange: FC<FilterSliderPropst> = ({ name, ...props }) => {
    const range = useAppSelector(state => state.filters.ranges.find(range => range.name === name)) as IFilterRange
    const [context, dispatch] = useCreateRangeContext(range)
    const { defaultValues, step } = range
    const { min, max } = defaultValues;
    const title = stringFirstLetterUppercase(name)
    const onChangeSlider = useChangeSliderValue(context, dispatch)
    const { id } = useParams()
    useEffect(() => {
        dispatch({ type: 'setBoth', payload: [min, max] })
    }, [id])
    return (
        <RangeContext.Provider value={[context, dispatch]}>
            <div className={s['container']}>
                <h2 className={s['title']}>{title}</h2>
                <Range
                    {...props}
                    min={min}
                    max={max}
                    defaultValue={[min, max]}
                    step={step}
                    range
                    {...styles}
                    value={[context.currentValues.min, context.currentValues.max]}
                    onChange={onChangeSlider}

                />
                <div className={s['inputs']}>
                    <div className={s['label']}>
                        <FilterRangeInput
                            title={'min'}
                            maxLength={3}
                            inputKey={'min'}
                        />
                    </div>
                    <span className={s['line']}>-</span>
                    <div className={s['label']}>
                        <FilterRangeInput
                            title={'max'}
                            maxLength={3}
                            inputKey={'max'}
                        />
                    </div>
                </div>
            </div>
        </RangeContext.Provider>
    );
};

export default memo(FilterRange);
