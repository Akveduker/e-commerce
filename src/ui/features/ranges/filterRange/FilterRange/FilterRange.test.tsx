import { fireEvent, screen } from "@testing-library/react"
import { withMemoryRouterAndRedux } from "../../../../../helper/testFc/withMemoryRouterAndRedux/withMemoryRouterAndRedux"
import FilterRange from "./FilterRange"

describe('FilterRange', () => {
    //имя ренджа в фейковом сторе price
    const element = <FilterRange name="price" />
    test('Снапшот', () => {
        const { container } = withMemoryRouterAndRedux(
            element
        )
        expect(container).toMatchSnapshot()
    })
})