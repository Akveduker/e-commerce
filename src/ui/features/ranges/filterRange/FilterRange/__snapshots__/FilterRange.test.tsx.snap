// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterRange Снапшот 1`] = `
<div>
  <div
    class="container"
  >
    <h2
      class="title"
    >
      Price
    </h2>
    <div
      class="rc-slider rc-slider-horizontal"
      style="display: flex; align-items: center; max-width: 246px; margin: 0px 10px;"
    >
      <div
        class="rc-slider-rail"
        style="background-color: rgb(235, 235, 235); border-radius: 12px; height: 6px; width: 266px; left: -10px;"
      />
      <div
        class="rc-slider-track rc-slider-track-1"
        style="left: 30%; width: 50%; background-color: rgb(106, 152, 60); height: 6px;"
      />
      <div
        class="rc-slider-step"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="30"
        class="rc-slider-handle rc-slider-handle-1"
        role="slider"
        style="left: 30%; transform: translateX(-50%); background-color: rgb(255, 255, 255); height: 20px; width: 20px; opacity: 1; border: 1px solid #d1d1d1; margin-top: 0px; box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.15);"
        tabindex="0"
      />
      <div
        aria-disabled="false"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="80"
        class="rc-slider-handle rc-slider-handle-2"
        role="slider"
        style="left: 80%; transform: translateX(-50%); background-color: rgb(255, 255, 255); height: 20px; width: 20px; opacity: 1; border: 1px solid #d1d1d1; margin-top: 0px; box-shadow: 0px 2px 4px rgba(0, 0, 0, 0.15);"
        tabindex="0"
      />
    </div>
    <div
      class="inputs"
    >
      <div
        class="label"
      >
        <label>
          <h4
            class="title"
          >
            min
          </h4>
          <input
            class="input"
            maxlength="3"
            type="text"
            value="30"
          />
        </label>
      </div>
      <span
        class="line"
      >
        -
      </span>
      <div
        class="label"
      >
        <label>
          <h4
            class="title"
          >
            max
          </h4>
          <input
            class="input"
            maxlength="3"
            type="text"
            value="80"
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
